src/wp-includes/block-template.php

```diff
@@ -224,6 +225,14 @@ function get_the_block_template_html() {
 		return;
 	}
 
+	/*
+	 * Check whether the current template actually comes from the theme.
+	 * This is generally the case, except when plugins hijack the block template loading process
+	 * to inject a custom template coming from a plugin.
+	 */
+	$current_theme_owns_current_template = $_wp_current_template_id &&
+		str_starts_with( $_wp_current_template_id, get_stylesheet() . '//' );
+
```

Comment:
Curious, not a blocker: Why so high up in the function rather than after `do_shortcode()`?

Reply:
Good point, though I am going with the suggestion by @mukeshpanchal27 in https://github.com/WordPress/wordpress-develop/pull/5575#discussion_r1372570507 instead, as that simplifies things.